### PR TITLE
Check if content has an identity before attempting to get it's ancestors

### LINKED ISSUE #12732: Get error on cancel saving document type
### Which *exact* Umbraco version are you using? For example: 9.0.1 - don't just write v9

10.0.1

### Bug summary

I create the document type which alias name as simplehomePage ,set as child of document type  and set this document type Allow vary by culture is true.
and then I create PageValidationNotificationHandler

Under Handler Method just make this code:
```
if (content.ContentType.Alias == "simplehomePage") {
			notification.CancelOperation(
				new EventMessage(
					"Publishing error",
					"You cannot save and publish this Home page",
					EventMessageType.Error));
		}
```
When I press save and publish , I've the error as below :
An error occurred
Input string was not in a correct format.

Stacktrace
```
at System.Number.ThrowOverflowOrFormatException(ParsingStatus status, TypeCode type)
   at System.Int32.Parse(String s, IFormatProvider provider)
   at Umbraco.Extensions.ContentExtensions.<>c.<GetAncestorIds>b__8_1(String s)
   at System.Linq.Enumerable.WhereSelectArrayIterator`2.MoveNext()
   at Umbraco.Cms.Web.BackOffice.Controllers.ContentController.GetPublishedCulturesFromAncestors(IContent content)
   at Umbraco.Cms.Web.BackOffice.Controllers.ContentController.AddDomainWarnings(IContent persistedContent, String[] culturesPublished, SimpleNotificationModel globalNotifications)
   at Umbraco.Cms.Web.BackOffice.Controllers.ContentController.PostSaveInternal[TVariant](ContentItemSave contentItem, Func`3 saveMethod, Func`2 mapToDisplay)
   at Umbraco.Cms.Web.BackOffice.Controllers.ContentController.PostSave(ContentItemSave contentItem)
   at lambda_method1429(Closure , Object )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.AwaitableObjectResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextExceptionFilterAsync>g__Awaited|26_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
```

But if I turn "Allow vary by culture" to false, it's work

### Specifics

_No response_

### Steps to reproduce

In settings,
Create the document type in this setting
Step 1:
image

step 2:
image

Step 3:
Create ContentSavingNotification as below code 

```
using Umbraco.Cms.Core.Events;
using Umbraco.Cms.Core.Notifications;
namespace UmbracoLite.web.Infrastructure.Handler;
public class PageValidationNotificationHandler : INotificationHandler<ContentSavingNotification> {
	public void Handle(ContentSavingNotification notification) {
		var content = notification.SavedEntities.FirstOrDefault();
		if (content.ContentType.Alias == "simpleHomePage") {
			notification.CancelOperation(
				new EventMessage(
					"Publishing error",
					"You cannot save and publish this Home page",
					EventMessageType.Error));
		}
	}
}
```
and then add code at startup configureservices
```
services.AddUmbraco(_env, _config)
			.AddBackOffice()
			.AddWebsite()
			.AddComposers()
			.AddNotificationHandler<ContentSavingNotification, PageValidationNotificationHandler>()
			.Build();

```

Step 4:
image
create the parent document type , and under parent create the child document type 
and then click save and publish
image

### Expected result / actual result

We expected user will show this error as below
image

but it throw ThrowOverflowOrFormatException

### PR DESCRIPTION
**Pull request by nul800sebastiaan**
This fixes #12732 

I'm not entirely sure if this is the correct fix, but it prevents the `OverflowOrFormatException` and it still gets all the cultures from the ancestors. 
The other fix I tried was to add `StringSplitOptions.RemoveEmptyEntries` in `GetAncestorIds`:

```csharp
 public static IEnumerable<int>? GetAncestorIds(this IContent content) =>
        content.Path?.Split(Constants.CharArrays.Comma, StringSplitOptions.RemoveEmptyEntries)
```

This works, but it would just return an empty collection. The fix in this PR still gets the cultures of the ancestors.

### CODE REVIEW

**Comment by nul800sebastiaan** [comment]
Interesting that this code is ONLY needed when someone is hooking into `ContentSavingNotification`, when there is no such notification being listened for, the content item has an identity when we get to the `GetPublishedCulturesFromAncestors` method.

File: src/Umbraco.Web.BackOffice/Controllers/ContentController.cs [hunk 1]
```diff
@@ -1797,6 +1797,11 @@ private bool PublishCulture(IContent persistentContent, IEnumerable<ContentVaria
 
     private IEnumerable<string> GetPublishedCulturesFromAncestors(IContent? content)
     {
+        if (content?.ParentId != -1 && content?.HasIdentity == false)
```
Comment (Zeegaan) [inline src/Umbraco.Web.BackOffice/Controllers/ContentController.cs:1800]:
```suggestion
        if (content?.ParentId is not -1 && content?.HasIdentity is false)
```
Nitpicking but pattern matching is the new hotness 🔥

Comment (nul800sebastiaan) [reply src/Umbraco.Web.BackOffice/Controllers/ContentController.cs]: Haha I forgot.. excellent! 👍

**Comment by Zeegaan** [comment]
Tests out great ✅ 
LGTM!

**Comment by Zeegaan** [comment]
Cherry picked for 10.1

**Pull request merged**